core/shared/src/main/scala/zio/Chunk.scala

```diff
@@ -2470,6 +2471,9 @@ object Chunk extends ChunkFactory with ChunkPlatformSpecific {
   final case class AnyRefArray[A <: AnyRef](array: Array[A], offset: Int, override val length: Int)
       extends Arr[A]
       with ChunkIterator[A] { self =>
+
+    val classTag: ClassTag[A] = Tags.fromValue(array(0))
```

Comment:
The previous implementation was inherited from `Arr` and was:
```scala
implicit val classTag: ClassTag[A] =
      ClassTag(array.getClass.getComponentType)
```

See https://github.com/zio/zio/pull/10079/files#diff-692b5e51eb20ff6846be420c52c1b9830e2f756b69e280e48234410676013254L1597-L1598

Not sure which one to prefer